### PR TITLE
[Mellanox] add Mellanox-SN4700-O28 SKU

### PR DESCRIPTION
**Pull request by Yakiv-Huryk**
#### Why I did it
To add new SKU for Virtual Smart Switch. T1 switch with 28x400G ports.

##### Work item tracking
- Microsoft ADO **(number only)**:

#### How I did it
Add new SKU with all relevant files.

#### How to verify it
run sonic-mgmt t1-28 test suites based on master. 
Few issues observed not relevant to the topology but to the stability of master

#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012
- [ ] 202106
- [ ] 202111
- [ ] 202205
- [ ] 202211
- [ ] 202305

#### Tested branch (Please provide the tested image version)

- [ ] 
- [ ]

### CODE REVIEW

**Review by dgsudharsan — APPROVED** [review]

**Review by oleksandrivantsiv — APPROVED** [review]

File: device/mellanox/x86_64-mlnx_msn4700-r0/Mellanox-SN4700-O28/port_config.ini [hunk 1]
```diff
@@ -0,0 +1,49 @@
+##
+## Copyright (c) 2020-2023 NVIDIA CORPORATION & AFFILIATES.
```
Comment (liat-grozovik) [inline device/mellanox/x86_64-mlnx_msn4700-r0/Mellanox-SN4700-O28/port_config.ini:2]: this is a new file thus license should be 2023 only.

File: device/mellanox/x86_64-mlnx_msn4700-r0/Mellanox-SN4700-O28/port_config.ini [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+## Apache-2.0
```
Comment (liat-grozovik) [inline device/mellanox/x86_64-mlnx_msn4700-r0/Mellanox-SN4700-O28/port_config.ini:3]: this is Nvidia specific file. License should be Nvidia one.

File: device/mellanox/x86_64-mlnx_msn4700-r0/Mellanox-SN4700-O28/sai.profile [hunk 3]
```diff
@@ -0,0 +1,3 @@
+SAI_INIT_CONFIG_FILE=/usr/share/sonic/hwsku/sai_4700.xml
+SAI_DUMP_STORE_PATH=/var/log/mellanox/sdk-dumps
+SAI_DUMP_STORE_AMOUNT=10
```
Comment (liat-grozovik) [inline device/mellanox/x86_64-mlnx_msn4700-r0/Mellanox-SN4700-O28/sai.profile:3]: Please sync with the recent changes of another SN4700 of store and forward and update it. check as a reference the changes done in https://github.com/sonic-net/sonic-buildimage/pull/16781

**Review by liat-grozovik — CHANGES_REQUESTED** [review]

File: device/mellanox/x86_64-mlnx_msn4700-r0/Mellanox-SN4700-O28/port_config.ini (hunk 2 above)
Comment (Yakiv-Huryk) [reply device/mellanox/x86_64-mlnx_msn4700-r0/Mellanox-SN4700-O28/port_config.ini:3]: Fixed in https://github.com/sonic-net/sonic-buildimage/pull/16784/commits/64761e9ad3f008c9ed3fd1e657b6121622fcb174

File: device/mellanox/x86_64-mlnx_msn4700-r0/Mellanox-SN4700-O28/sai.profile (hunk 3 above)
Comment (Yakiv-Huryk) [reply device/mellanox/x86_64-mlnx_msn4700-r0/Mellanox-SN4700-O28/sai.profile:3]: Fixed in https://github.com/sonic-net/sonic-buildimage/pull/16784/commits/64761e9ad3f008c9ed3fd1e657b6121622fcb174

File: device/mellanox/x86_64-mlnx_msn4700-r0/Mellanox-SN4700-O28/port_config.ini (hunk 1 above)
Comment (Yakiv-Huryk) [reply device/mellanox/x86_64-mlnx_msn4700-r0/Mellanox-SN4700-O28/port_config.ini]: Fixed in https://github.com/sonic-net/sonic-buildimage/pull/16784/commits/64761e9ad3f008c9ed3fd1e657b6121622fcb174

**Review by liat-grozovik — APPROVED** [review]

**Pull request merged**